docs/code/MlNetCookBook.md

```diff
@@ -447,6 +532,46 @@ var learningPipeline = reader.MakeNewEstimator()
 var model = learningPipeline.Fit(trainData).AsDynamic;
 ```
 
+You can also use the dynamic API to create the equivalent of the previous pipeline. 
+```csharp
+// Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+// as a catalog of available operations and as the source of randomness.
+var mlContext = new MLContext();
+
+// Step one: read the data as an IDataView.
+// First, we define the reader: specify the data columns and where to find them in the text file.
+var reader = mlContext.Data.TextReader(new TextLoader.Arguments
+{
+    Column = new[] {
+        new TextLoader.Column("SepalLength", DataKind.R4, 0),
+        new TextLoader.Column("SepalWidth", DataKind.R4, 1),
+        new TextLoader.Column("PetalLength", DataKind.R4, 2),
+        new TextLoader.Column("PetalWidth", DataKind.R4, 3),
+        // Label: kind of iris.
+        new TextLoader.Column("Label", DataKind.TX, 4),
+    },
+    // Default separator is tab, but the dataset has comma.
+    Separator = ","
+});
+
+// Retrieve the training data.
+var trainData = reader.Read(irisDataPath);
+
+// Build the training pipeline.
+var dynamicPipeline =
+    // Concatenate all the features together into one column 'Features'.
+    mlContext.Transforms.Concatenate("Features", "SepalLength", "SepalWidth", "PetalLength", "PetalWidth")
+    // Note that the label is text, so it needs to be converted to key.
+    .Append(new ValueToKeyMappingEstimator(mlContext, "Label"), TransformerScope.TrainTest)
```

Comment:
>ValueToKeyMappingEstimator [](start = 16, length = 26)

`mlContext.Transforms.Categorical.MapValueToKey`

Reply:
changed.

---
In reply to: [231225091](https://github.com/dotnet/machinelearning/pull/1538#discussion_r231225091) [](ancestors = 231225091)